Source/Core/Parents/ProfileLoader.cpp

```diff
@@ -78,11 +78,11 @@ namespace ProfileLoader3
     {
         Profile3 getProfile(const json &j)
         {
-            std::string name = j["name"].get<std::string>();
-            Game version = j["version"].get<Game>();
-            u16 tid = j["tid"].get<u16>();
-            u16 sid = j["sid"].get<u16>();
-            bool deadBattery = j["battery"].get<bool>();
+            std::string name = j.value("name", std::string("-"));
```

Comment:
Two question for this entire file.

1. I think it is unnecessary to change to .value() for fields that already exist
2. What is up with all the c-style casts? Can the templated function not pick up on the type of the default value?

Reply:
The casts are the same reason for the .get() from before. It uses the type of the second parameter when parsing. I understand some may not need the casting but I wasn't sure if C++ handled type safety like that automatically.

Reply:
Changing it for all of them just provides consistency. I understand some will never be missing unless deleted by the user but the safety is better to have than to not.

Reply:
I removed all the type casting and it built fine. I suspect the fact that the type is on the left hand of the assignment is enough to inform the compiler of the templated types.

Reply:
> I removed all the type casting and it built fine. I suspect the fact that the type is on the left hand of the assignment is enough to inform the compiler of the templated types.

removed all the type casting